addons/pos_event/static/src/app/components/popup/event_slot_selection_popup/event_slot_selection_popup.js

```diff
@@ -0,0 +1,72 @@
+// Part of Odoo. See LICENSE file for full copyright and licensing details.
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { Dialog } from "@web/core/dialog/dialog";
+import { Component, useState } from "@odoo/owl";
+import { formatDate } from "@web/core/l10n/dates";
+import { localization } from "@web/core/l10n/localization";
+import { NumericInput } from "@point_of_sale/app/components/inputs/numeric_input/numeric_input";
+import { ProductCard } from "@point_of_sale/app/components/product_card/product_card";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+
+export class EventSlotSelectionPopup extends Component {
+    static template = "pos_event.EventSlotSelectionPopup";
+    static props = ["getPayload", "close", "event"];
+    static components = {
+        Dialog,
+        ProductCard,
+        NumericInput,
+    };
+    setup() {
+        this.dialog = useService("dialog");
+        this.slotId = false;
+        this.state = useState({
+            selectedSlotDisplay: "",
+        });
+    }
+    get dialogTitle() {
+        return _t("Select a slot for %(event)s", { event: this.props.event.name });
+    }
+    get slots() {
+        const slots = {};
+        this.props.event.slot_ids.forEach((slot) => {
+            const date = formatDate(slot.date, { format: "MMM dd yyyy, EEEE" });
+            if (!slots[date]) {
+                slots[date] = {};
+            }
+            slots[date][slot.id] = slot.start_datetime.toFormat(
```

Comment:
here the date is tz naive and is the value as per event_tz. Any shift due to the POS session?